### DISCUSSION TITLE
Does Scala CLI support the workspace/reload BSP endpoint?

### DISCUSSION THREAD

**Discussion by julienrf** [VirtusLab/scala-cli#1659] [General]
Please see the detailed description here: https://youtrack.jetbrains.com/issue/SCL-20667/Re-importing-a-BSP-project-does-not-work

**Comment by tgodzik** [comment]
I commented on the issues. It's supported but it does nothing. The build definition should refresh automatically and we send a notification about it. @Gedochao Any idea what might be wrong? We had the same issue in Metals for sbt BSP

**Comment by Gedochao** [comment]
It actually doesn't do nothing, we should be supporting `workspace/reload`.
https://github.com/VirtusLab/scala-cli/blob/adca9c2555d15640b6effdcf6b6ec204fada63ce/modules/build/src/main/scala/scala/build/bsp/BuildServerProxy.scala#L80
https://github.com/VirtusLab/scala-cli/blob/01960bec2cd54f2cd7664a3ebb8a0190481e4747/modules/build/src/main/scala/scala/build/bsp/BspImpl.scala#L557

After testing it a bit just now, it technically does work, but in this use case it's a bit wonky.

https://user-images.githubusercontent.com/18601388/206666017-93d5e42c-479c-4598-8cd5-e24bc7b533a6.mov

after adding a new directive/dependency in the code, `workspace/reload` doesn't seem to pick it up until the first re-compilation is triggered. It doesn't matter if it's from the command line, or if it's a re-run from IntelliJ and the re-run should go through BSP).

Not sure where's the bug, but it's worth investigating on our side. I'll create a ticket.

**Reply by Gedochao** [reply]
can be tracked under https://github.com/VirtusLab/scala-cli/issues/1673

**Reply by Gedochao** [reply]
Should be fixed in #1681
